python/lsst/meas/astrom/fitAffineWcs.py

```diff
@@ -163,52 +116,41 @@ def fitWcs(self,
         # appends the new transform.
         wcsMaker = TransformedSkyWcsMaker(initWcs)
 
-        refPoints = []
-        srcPixels = []
-        offsetLong = 0
-        offsetLat = 0
-        # Grab reference coordinates and source centroids. Compute the average
-        # direction and separation between the reference and the sources.
-        for match in matches:
+        # Grab the initial transformations going back from sky coordinates
+        # and forward from pixel coordinates.
+        back = wcsMaker.frameDict.getMapping(wcsMaker.frameMax, wcsMaker.frameMax-1)
+        forward = wcsMaker.lastMapBeforeSky
+
+        # Create containers for the data going into the Affine fit (i.e.
+        # solving Ax=b where x will be the affine parameters and a linear
+        # shift.
+        A = np.zeros((len(matches)*2, 6), dtype=float)
+        b = np.empty(len(matches)*2, dtype=float)
+
+        # Constant terms related to the shit in and and y parameters
+        A[:len(matches)*2:2, 4] = 1
+        A[1:len(matches)*2+1:2, 5] = 1
+
+        # loop over each of the matches and populate the matrices
+        for i, match in enumerate(matches):
             refCoord = match.first.getCoord()
-            refPoints.append(refCoord)
+            b[i*2:i*2+2] = back.applyForward(refCoord)
+
             srcCentroid = match.second.getCentroid()
-            srcPixels.append(srcCentroid)
-            srcCoord = initWcs.pixelToSky(srcCentroid)
-            deltaLong, deltaLat = srcCoord.getTangentPlaneOffset(refCoord)
-            offsetLong += deltaLong.asArcseconds()
-            offsetLat += deltaLat.asArcseconds()
-        offsetLong /= len(srcPixels)
-        offsetLat /= len(srcPixels)
-        offsetDist = np.sqrt(offsetLong ** 2 + offsetLat ** 2)
-        if offsetDist > 0.:
-            offsetDir = np.degrees(np.arccos(offsetLong / offsetDist))
-        else:
-            offsetDir = 0.
-        offsetDir *= np.sign(offsetLat)
-        self.log.debug("Initial shift guess: Direction: %.3f, Dist %.3f...",
-                       offsetDir, offsetDist)
-
-        # Best performing fitter in scipy tried so far (vs. default settings in
-        # minimize). Exits early because of the xTol value which cannot be
-        # disabled in scipy1.2.1. Matrix starting values are non-zero as this
-        # results in better fit off-diagonal terms.
-        fit = least_squares(
-            _chiFunc,
-            x0=[offsetDir, offsetDist, 1., 1e-8, 1e-8, 1.],
-            args=(refPoints, srcPixels, wcsMaker),
-            method='dogbox',
-            bounds=[[-360, -np.inf, -np.inf, -np.inf, -np.inf, -np.inf],
-                    [360, np.inf, np.inf, np.inf, np.inf, np.inf]],
-            ftol=2.3e-16,
-            gtol=2.31e-16,
-            xtol=2.3e-16)
-        self.log.debug("Best fit: Direction: %.3f, Dist: %.3f, "
+            val = forward.applyForward(srcCentroid)
```

Comment:
I'm pretty sure we could get another speedup by vectorizing these calls to `applyForward` if the profile says that'd help. The usefulness of that is limited by the fact that we're starting with a `ReferenceMatchVector` rather than something with array-column coordinates, though.

Reply:
Yeah I thought about this too. Most of the time seems to be "within this function" so I suspect it is things like this. However it was so much faster I didn't know if it was worth opening up another repo and adding an interface which is pretty much just for this. I would be happy to do so if we thought it worth it.

Reply:
Let's just get this in.  Doing more could easily take a lot longer than this ticket has so far.